fuzzbench/run_experiment.py

```diff
@@ -18,20 +18,39 @@
 import redis
 import rq
 
+from common import config_utils, environment, yaml_utils
+from experiment.build import docker_images
 from fuzzbench import jobs
 
+redis_connection = redis.Redis(host="queue-server")  # pylint: disable=invalid-name
 
-def run_experiment():
+
+def run_experiment(config):
     """Main experiment logic."""
     print('Initializing the job queue.')
     # Create the queue for scheduling build jobs and run jobs.
     queue = rq.Queue('build_n_run_queue')
+
+    images_to_build = docker_images.get_images_to_build(config['fuzzers'],
+                                                        config['benchmarks'])
     jobs_list = []
-    jobs_list.append(
-        queue.enqueue(jobs.build_image,
-                      'base-image',
-                      job_timeout=600,
-                      job_id='base-image'))
+    for name, obj in images_to_build.items():
+        if 'depends_on' not in obj:
+            jobs_list.append(
+                queue.enqueue(jobs.build_image,
+                              job_obj=obj,
+                              job_timeout=1800,
+                              result_ttl=config['max_total_time'],
+                              job_id=name))
+            continue
```

Comment:
can we remove this part, and have a single enqueue() call in the for loop?

Reply:
Done.